### ISSUE TITLE
NPE from CategoryProvicer.getCategory(185) when called from LazyLibrarian

### ISSUE THREAD

**Issue by claym** [theotherp/nzbhydra2#129]
So, get this issue when calling from the latest LazyLibrarian to NzbHydra2 1.4.3

URL called: https://someurl/api?author=Jeff+VanderMeer&apikey=123456&t=book&cat=7000%2C7020&title=Annihilation

Hydra Logs: 

Received external API call: NewznabParameters{t=BOOK, cat=[7000, 7020], author=Jeff VanderMeer, title=Annihilation, raw=false, extended=false}

Unexpected error while handling API request

java.lang.NullPointerException: null
	at org.nzbhydra.searching.CategoryProvider.getCategory(CategoryProvider.java:185)
	at org.nzbhydra.searching.CategoryProvider.fromSearchNewznabCategories(CategoryProvider.java:144)
	at org.nzbhydra.api.ExternalApi.buildBaseSearchRequest(ExternalApi.java:316)
	at org.nzbhydra.api.ExternalApi.search(ExternalApi.java:256)
	at org.nzbhydra.api.ExternalApi.api(ExternalApi.java:96)

**Comment by theotherp** [comment]
Please post the debug infos ZIP. Is it so hard to follow the instructions?

**Comment by claym** [comment]
Perhaps move the "please provide debug logs" to the top instead of "you don't have to follow". I just skimmed over the rest when I saw that.

Debug logs attached.

[nzbhydra-debuginfos-2018-02-27-09-05.zip](https://github.com/theotherp/nzbhydra2/files/1763160/nzbhydra-debuginfos-2018-02-27-09-05.zip)

**Comment by theotherp** [comment]
> Perhaps move the "please provide debug logs" to the top instead of "you don't have to follow". I just skimmed over the rest when I saw that.

I don't even know how to respond to that.

I'll push a fix soonish. The error is caused by a very weird beahvior by LL. I'll contact its developer to find out why it does what it does.

**Comment by claym** [comment]
I was not trying to be a jerk, I was trying to explain what led to not following the instructions and offering a solution to make it a little more obvious.

**Comment by theotherp** [comment]
I've changed the template, thanks.

Am 27.02.2018 8:06 nachm. schrieb "Clay Mitchell" <notifications@github.com
>:

I was not trying to be a jerk, I was trying to explain what led to not
following the instructions and offering a solution to make it a little more
obvious.

—
You are receiving this because you modified the open/close state.

Reply to this email directly, view it on GitHub
<https://github.com/theotherp/nzbhydra2/issues/129#issuecomment-368990075>,
or mute the thread

**Comment by claym** [comment]
The fix you requested from the LL guys fixed things. Thanks!
